### PR TITLE
🚃 create pubapp

### PR DESCRIPTION
**Pull request by dmiller15**
## Description

This PR creates a new, standaridized public app for the RNAseq workflow. Small formatting changes were also made to clean up the readme such as removing empty lines and indentation fixing.

Closes https://github.com/d3b-center/bixu-tracker/issues/762

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] cwltool validation
- [x] Cavatica test: https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/cbeaf42e-913b-41ca-9163-f813cfd5ec4c/#

**Test Configuration**:
* Environment: cwltool 3.0.20200324120055, cavatica
* Test files: see cavatica test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: workflow/kfdrc-rnaseq-workflow.cwl [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+hints:
+- class: 'sbg:maxNumberOfParallelInstances'
+  value: 3
+sbg:license: Apache License 2.0
+sbg:publisher: KFDRC
+sbg:categories:
+- ALIGNMENT
+- ARRIBA
+- BAM
+- CRAM
```
Comment (migbro) [inline workflow/kfdrc-rnaseq-workflow.cwl:370]: Eh, I don't think so

File: workflow/kfdrc-rnaseq-workflow.cwl [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+    arriba_fusion_results: {type: File, outputSource: arriba_fusion/arriba_fusions}
+    arriba_fusion_viz: {type: File, outputSource: arriba_fusion/arriba_pdf}
+    RSEM_isoform: {type: File, outputSource: rsem/isoform_out}
+    RSEM_gene: {type: File, outputSource: rsem/gene_out}
+    RNASeQC_Metrics: {type: File, outputSource: rna_seqc/Metrics}
+    RNASeQC_counts: {type: File, outputSource: supplemental/RNASeQC_counts} # contains gene tpm, gene read, and exon counts
+    kallisto_Abundance: {type: File, outputSource: kallisto/abundance_out}
+  ```
+
+  # D3b annoFuse Workflow
```
Comment (migbro) [inline workflow/kfdrc-rnaseq-workflow.cwl:131]: The downside of importing the README as-is, that the repo itself may have more than the app. Please remove this part from the RNAseq workflow `doc`

File: workflow/kfdrc-rnaseq-workflow.cwl (hunk 1 above)
Comment (dmiller15) [reply workflow/kfdrc-rnaseq-workflow.cwl]: You are correct. Fixed.

File: workflow/kfdrc-rnaseq-workflow.cwl (hunk 2 above)
Comment (dmiller15) [reply workflow/kfdrc-rnaseq-workflow.cwl]: Removed.

File: workflow/kfdrc-rnaseq-workflow.cwl [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+  ![data service logo](https://github.com/d3b-center/d3b-research-workflows/raw/master/doc/kfdrc-logo-sm.png)
+
+  ## Introduction
+  This pipeline utilizes cutadapt to trim adapters from the raw reads, if necessary, and passes the reads to STAR for alignment.
+  The alignment output is used by RSEM for gene expression abundance estimation.
+  Additionally, Kallisto is used for quantification, but uses pseudoalignments to estimate the gene abundance from the raw data.
+  Fusion calling is performed using Arriba and STAR-Fusion detection tools on the STAR alignment outputs.
+  Metrics for the workflow are generated by RNA-SeQC.
+  **A new feature added as of February 27, 2020**, is an added workflow to run the artifact filtering portion of [annoFuse](https://www.biorxiv.org/content/10.1101/839738v1).
```
Comment (migbro) [inline workflow/kfdrc-rnaseq-workflow.cwl:16]: perhaps remove this reference here...we should talk to Krutika and JoLynne about making annoFuse a Cavatica public app though...

File: README.md [hunk 4]
```diff
@@ -4,10 +4,6 @@ This is the Kids First RNA-Seq pipeline, which includes fusion and expression de
 
 ![data service logo](https://github.com/d3b-center/d3b-research-workflows/raw/master/doc/kfdrc-logo-sm.png)
 
-
-
-
-
 ## Introduction
 This pipeline utilizes cutadapt to trim adapters from the raw reads, if necessary, and passes the reads to STAR for alignment.
 The alignment output is used by RSEM for gene expression abundance estimation.
```
Comment (migbro) [inline README.md:9]: Similar to joint JG comment, perhaps add a link to the public app here, and in the doc of the cwl, add a link to the git release

**Review by migbro — CHANGES_REQUESTED** [review]
Two small things to address before approving.

File: README.md (hunk 4 above)
Comment (dmiller15) [reply README.md:9]: Similar to the JG workflow my solution was to add a link to the public app in the readme as you did but I avoided adding it to the doc and instead made it an sbg:links label.

File: workflow/kfdrc-rnaseq-workflow.cwl (hunk 3 above)
Comment (dmiller15) [reply workflow/kfdrc-rnaseq-workflow.cwl]: Removed the two feature update notes.

File: README.md [hunk 5]
```diff
@@ -4,20 +4,16 @@ This is the Kids First RNA-Seq pipeline, which includes fusion and expression de
...
 Metrics for the workflow are generated by RNA-SeQC.
-**A new feature added as of February 27, 2020**, is an added workflow to run the artifact filtering portion of [annoFuse](https://www.biorxiv.org/content/10.1101/839738v1).
-Developed collaboratively between the Center for Data Driven Discovery in Biomedicine (D3b) and the Alex's Lemonade Stand Childhood Cancer Data Lab, this package adds annotation to arriba results, artifact filtering, and removes low-confidence fusion calls, as described in the paper.
-There are additional filtering capabilities provided beyond the lightweight workflow described in the [D3b annoFuse Workflow](#d3b-annofuse-workflow) section.
-**A new feature added as of May 21, 2020**, we now accept single end fastq inputs.
+
+Inputs from this can be used later on for further analysis in joint trio genotyping and subsequent refinement and deNovo variant analysis. If you would like to run this workflow using the cavatica public app, a basic primer on running public apps can be found [here](https://www.notion.so/d3b/Starting-From-Scratch-Running-Cavatica-af5ebb78c38a4f3190e32e67b4ce12bb).
```
Comment (migbro) [inline README.md:14]: That first sentence should also go, just noticed that.

File: workflow/kfdrc-rnaseq-workflow.cwl [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+  ![data service logo](https://github.com/d3b-center/d3b-research-workflows/raw/master/doc/kfdrc-logo-sm.png)
+
+  ## Introduction
+  This pipeline utilizes cutadapt to trim adapters from the raw reads, if necessary, and passes the reads to STAR for alignment.
+  The alignment output is used by RSEM for gene expression abundance estimation.
+  Additionally, Kallisto is used for quantification, but uses pseudoalignments to estimate the gene abundance from the raw data.
+  Fusion calling is performed using Arriba and STAR-Fusion detection tools on the STAR alignment outputs.
+  Metrics for the workflow are generated by RNA-SeQC.
+
+  Inputs from this can be used later on for further analysis in joint trio genotyping and subsequent refinement and deNovo variant analysis. If you would like to run this workflow using the cavatica public app, a basic primer on running public apps can be found [here](https://www.notion.so/d3b/Starting-From-Scratch-Running-Cavatica-af5ebb78c38a4f3190e32e67b4ce12bb).
```
Comment (migbro) [inline workflow/kfdrc-rnaseq-workflow.cwl:17]: Same as readme, this first sentence should go.

**Review by migbro — CHANGES_REQUESTED** [review]
So close! Missed another out-of-place sentence before, sorry about that.

File: workflow/kfdrc-rnaseq-workflow.cwl (hunk 6 above)
Comment (dmiller15) [reply workflow/kfdrc-rnaseq-workflow.cwl]: Good catch. Removed.

File: README.md (hunk 5 above)
Comment (dmiller15) [reply README.md]: Good catch. Removed.

**Review by migbro — APPROVED** [review]
and another one

**Pull request merged**